docs/security/workbenchAssets.md

```diff
@@ -0,0 +1,93 @@
+
+```
+# Manager role only have access to the HomeForManagers perspective
+role.manager.permission.perspective.read=false
+role.manager.permission.perspective.read.HomeForManagers=true
+```
+
+The above is an example of a `deny all grant a few` strategy.
+
+Next is an example of the opposite `grant all deny a few` strategy:
+
+```
+# Admin role users have access to all the perspectives but the HomeForManagers
+role.admin.permission.perspective.read=true
+role.admin.permission.perspective.read.HomeForManagers=false
+```
+
+As shown above the policy is flexible enough to model the most common scenarios. Besides, it can be used to protect other workbench assets. For instance:
+
+```
+# A superuser role can read everything
+role.superuser.permission.perspective.read=true
+role.superuser.permission.screen.read=true
+role.superuser.permission.popup.read=true
+role.superuser.permission.splash.read=true
+role.superuser.permission.editor.read=true
+```
+
+#### Menu items
+
+Menus can be used to implement nagivation features across the workbench. It is possible to define menu items linked to perspectives. For instance, consider the following menu structure which might be perfectly an example of the main menu of a typical Uberfire app:
+
+```
+import static org.uberfire.workbench.model.menu.MenuFactory.*;
+...
+    @Inject
+    private WorkbenchMenuBar menubar;
+
+    private void setupMenu(@Observes final ApplicationReadyEvent event) {
+        final Menus menus =
+        newTopLevelMenu("Tasks")
+            .perspective("TasksPerspective")
+            .endMenu()
+        .newTopLevelMenu("Dashboard")
+            .perspective("DashboardPerspective")
+            .endMenu()
+            .build();
+
+        menubar.addMenus( menus );
+    }
+...
+```
+
+We have 2 menu entries visible in this app's top level menu, both linked to perspectives. The method `perspective(...)` it is used to set the target perspective identifier the system will go to after clicking on the menu item. Every menu item will show/hide depending on the perspective access permission granted to the user.
+
+For non perspective-linked menu items another extra mechanism is provided. Imagine that a third menu entry is added:
+
+```
+        .newTopLevelMenu("About")
+            .identifier("uftasks.about")
+            .respondsWith(() -> Window.alert( "Tasks App 1.0.0" ))
+            .endMenu()
+```
+
+Pay special attention to the method `identifier(...)` which is used to assign a unique id. to the menu item. Once the identifier is set, new permissions can be inserted into the security policy:
+
+```
+role.manager.permission.uftasks.about=false
+```
+
+This identifier setting approach also works with any workbench asset, just by means of appending the asset type preffix together with the asset identifier as the menu item identifier. For example:
```

Comment:
The permission name here can be any string. The dots are just testimonial.

Anyway, this chapter has changed slightly so I suggest doing a 2nd review.